src/assertions/descendants.js

```diff
@@ -1,7 +1,7 @@
 export default function descendants ({ wrapper, markup, arg1, sig, flag }) {
   const exactlyCount = flag(this, 'exactlyCount')
 
-  if (exactlyCount) {
+  if (typeof exactlyCount === 'number') {
```

Comment:
this should probably also do `&& isFinite(exactlyCount)` to cover `NaN` and exclude ±`Infinity`